### PR TITLE
Unittests2.0

### PR DESCRIPTION
**Pull request by keefehuang**
New pull request for integration tests for openfoam adapter.

### CODE REVIEW

**Comment by huangq1234553** [comment]
Integration tests created to perform run-time checks on data passed to preCICE.
Tests based off tutorial case already in openfoam adapter repository.
Tests are created for openfoam5 and require FakeIt header only mocking library.

Dependencies:
FakeIt v2.0.5: Download release version from github repository and untar.
```
wget https://github.com/eranpeer/FakeIt/archive/2.0.5.tar.gz
tar -xvg 2.0.5.tar.gz
```
Building Integration Test:
1) Ensure FakeIt standalone header file is in CPLUS_INCLUDE_PATH
```
export CPLUS_INCLUDE_PATH=/path/to/fakeit/single_header/standalone:$CPLUS_INCLUDE_PATH
```
2) Source openfoam environment
```
source /path/to/openfoam/etc/bashrc
```
3) Run ./Allwmake in /Tests/IntegrationTests/ in openfoam-adapter repository, unittests2.0 branch
```
cd ./Tests/IntegrationTests
./Allwmake
```
4) Run Fluid solver in /tutorials/CHT/flow-over-plate/buoyantPimpleFoam-laplacianFoam in openfoam-adapter repository, unittests2.0 branch
```
cd ./tutorials/CHT/flow-over-plate/buoyantPimpleFoam-laplacianFoam
./runFluid
```
The tests will then execute.

**Comment by huangq1234553** [comment]
Installed and ran test on Virtual Machine. Should be good to go!

File: Adapter.H [hunk 1]
```diff
@@ -1,25 +1,48 @@
 #ifndef PRECICEADAPTER_H
 #define PRECICEADAPTER_H
 
-#include "Interface.H"
+#ifdef UNIT_TESTING
+    #include "Tests/UnitTests/AdapterTest/AdapterUnittestIncludes.h"
```
Comment (MakisH) [inline Adapter.H:5]: @huangq1234553 what happened to the `Tests/UnitTests` directory? Are these exactly the same files as in #118?

**Comment by MakisH** [comment]
This is quite behind to be merged without rewriting a large part, but the idea is very nice. I close it for now, keep the branch, and may return to it in the future to rethink how we want to do unit tests in the adapter.

**Pull request closed without merging**